### ISSUE TITLE
If possible add a notice to stat generation, if undefined elements are used.

### ISSUE THREAD

**Issue by svenoe** [RotherOSS/otobo#3792]
In certain conditions within `_GenerateDynamicStats()` of `Kernel/System/Stats.pm` undefined elements can lead to warnings flooding the log or even sending emails if the Daemon is generating stats. This (I believe!) happens in one of the elements like https://github.com/RotherOSS/otobo/blob/50ff425f8ffc0f77319770ffbd2fbec4cdca1903/Kernel/System/Stats.pm#L3220 and to my knowledge can be triggered by e.g. defining a queue, adding it to a stat in a certain way, and afterwards setting it as invalid or renaming it or so. (Not sure exactly.)

Please have a look whether you can think of a case here, and test it, and if something indeed is undefined, test that prior and write a notice to the log containing the StatID. (Ask @KhalilHamada-OTOBO for more info.)

**Comment by stefanhaerter** [comment]
This is reproducible if integrating a dynamic field into a dynamic statistic and setting it to 'Invalid' afterwards. I configured the dynamic field as 'OrderBy', and after setting it to invalid, I observe the following log messages:

```
[Wed Sep 25 14:55:42 2024] -e: Use of uninitialized value in hash element at /opt/otobo/bin/psgi-bin/../../Kernel/System/Stats.pm line 3228.
[Wed Sep 25 14:55:42 2024] -e: Use of uninitialized value in string comparison (cmp) at /opt/otobo/bin/psgi-bin/../../Kernel/System/Stats/Dynamic/TicketList.pm line 1983.
[Wed Sep 25 14:55:42 2024] -e: Use of uninitialized value in string comparison (cmp) at /opt/otobo/bin/psgi-bin/../../Kernel/System/Stats/Dynamic/TicketList.pm line 1983.
[Wed Sep 25 14:55:42 2024] -e: Use of uninitialized value in string comparison (cmp) at /opt/otobo/bin/psgi-bin/../../Kernel/System/Stats/Dynamic/TicketList.pm line 1983.
[Wed Sep 25 14:55:42 2024] -e: Use of uninitialized value in string comparison (cmp) at /opt/otobo/bin/psgi-bin/../../Kernel/System/Stats/Dynamic/TicketList.pm line 1983.
[Wed Sep 25 14:55:42 2024] -e: Use of uninitialized value in string comparison (cmp) at /opt/otobo/bin/psgi-bin/../../Kernel/System/Stats/Dynamic/TicketList.pm line 1983.
[Wed Sep 25 14:55:42 2024] -e: Use of uninitialized value in string comparison (cmp) at /opt/otobo/bin/psgi-bin/../../Kernel/System/Stats/Dynamic/TicketList.pm line 1983.
[Wed Sep 25 14:55:42 2024] -e: Use of uninitialized value in string comparison (cmp) at /opt/otobo/bin/psgi-bin/../../Kernel/System/Stats/Dynamic/TicketList.pm line 1983.
[Wed Sep 25 14:55:42 2024] -e: Use of uninitialized value in string comparison (cmp) at /opt/otobo/bin/psgi-bin/../../Kernel/System/Stats/Dynamic/TicketList.pm line 1983.
```

**Comment by stefanhaerter** [comment]
Closed with merged #3844
